plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/business/internal/query/DNodeQuery.java

```diff
@@ -81,4 +88,30 @@ public Dimension getDefaultDimension() {
         return result;
     }
 
+    /**
+     * Return {@code true} if the specified DNode is in "auto size" mode, i.e. has a Size Computation Expression set to
+     * -1 or is empty; {@code false} otherwise.
+     * 
+     * @return {@code true} if the specified DNode is in "auto size" mode, i.e. has a Size Computation Expression set to
+     *         -1 or is empty; {@code false} otherwise.
+     */
+    public boolean isAutoSizeStyle() {
+        boolean isAutoSizeStyle = false;
+        if (node.getStyle() != null && node.getStyle().getDescription() instanceof NodeStyleDescription nodeStyleDesc && nodeStyleDesc.getLabelPosition().getValue() == LabelPosition.NODE) {
+            IInterpreter interpreter = SiriusPlugin.getDefault().getInterpreterRegistry().getInterpreter(node);
+            String sizeComputationExpression = nodeStyleDesc.getSizeComputationExpression();
```

Comment:
We should change the tooltip documentation in the VSM "expected Interger" but now we should also add -1 for autosize

Reply:
Tests and docs will be added once this auto-size approach has been validated.